src/components/headerSearch.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from "react";
+import Mark from "mark.js";
+import { IonButton, IonButtons, IonIcon, IonInput, IonItemDivider } from "@ionic/react";
+import { arrowDown, arrowUp } from 'ionicons/icons';
+
+const HeaderSearch = () => {
+  const [needle, setNeedle] = useState<string>('');
+  const [marks, setMarks] = useState<NodeListOf<HTMLElement>>(document.querySelectorAll('mark'));
+  const [currentIndex, setCurrentIndex] =  useState<number>(0);
+  const doc = document.querySelector('.akoma-ntoso') as HTMLElement;
```

Comment:
This needs to be passed in somehow, so that wherever HeaderSearch is used, it knows to search the page that it's included with. Maybe a method must be called externally by the embedded component after it's mounted and refs are valid?